packages/vue3/src/useForm.ts

```diff
@@ -31,27 +31,27 @@ interface InertiaFormProps<TForm extends Record<string, unknown>> {
 
 export type InertiaForm<TForm extends Record<string, unknown>> = TForm & InertiaFormProps<TForm>
 
-export default function useForm<TForm extends Record<string, unknown>>(data: TForm): InertiaForm<TForm>
+export default function useForm<TForm extends Record<string, unknown>>(data: TForm | (() => TForm)): InertiaForm<TForm>
 export default function useForm<TForm extends Record<string, unknown>>(
   rememberKey: string,
-  data: TForm,
+  data: TForm | (() => TForm),
 ): InertiaForm<TForm>
 export default function useForm<TForm extends Record<string, unknown>>(
-  rememberKeyOrData: string | TForm,
-  maybeData?: TForm,
+  rememberKeyOrData: string | TForm | (() => TForm),
+  maybeData?: TForm | (() => TForm),
 ): InertiaForm<TForm> {
   const rememberKey = typeof rememberKeyOrData === 'string' ? rememberKeyOrData : null
-  const data = typeof rememberKeyOrData === 'object' ? rememberKeyOrData : maybeData
+  const data = typeof rememberKeyOrData === 'string' ? maybeData : rememberKeyOrData
   const restored = rememberKey
     ? (router.restore(rememberKey) as { data: TForm; errors: Record<keyof TForm, string> })
     : null
-  let defaults = cloneDeep(data)
+  let defaults = typeof data === 'object' ? cloneDeep(data) : cloneDeep(data())
   let cancelToken = null
   let recentlySuccessfulTimeoutId = null
   let transform = (data) => data
 
   let form = reactive({
-    ...(restored ? restored.data : data),
```

Comment:
Note that this is a breaking change: previously, it was possible to pass in a reactive object like this:

```js
const product = reactive({ name: '' });

const form = useForm({ product });
```

This way, `product.name` and `form.product.name` would always be in sync.

However, this no longer works, as `data` is now deep-cloned twice before spreading into the form object.